### ISSUE TITLE
Update Raygui to version 4.0

### ISSUE THREAD

**Issue by leotada** [schveiguy/raylib-d#38]
https://github.com/raysan5/raygui/releases/tag/4.0

**Comment by schveiguy** [comment]
Thanks for reminding me. I will take a look!

**Comment by LiamM32** [comment]
If I did this, how would I go about it?

It looks like the current `raygui.d` was made by copying `raygui.h`, and then making edits until it's valid in D.

It looks like the file in this repository doesn't have the old `raygui.h` as part of it's git history. But should I somehow take advantage of git's features to add all of the new lines from the newer `raygui.h` into the current one, while excluding the lines that were never placed in the D version? Or should I just compare them by eye and try making the changes to `raygui.d` to mimic the functions in Ragyui 4.0?

**Comment by schveiguy** [comment]
What I did was use [ctod](https://github.com/dkorpel/ctod) and then just fixed what didn't compile. I don't think there's an "upgrade" path, you just have to do it again.

**Comment by LiamM32** [comment]
> What I did was use [ctod](https://github.com/dkorpel/ctod) and then just fixed what didn't compile. I don't think there's an "upgrade" path, you just have to do it again.

I'm trying to do an upgrade process where I placed `raygui.h` in a new git repository, and then made changes to it in multiple branches to reach both you're version of `raygui.d`, and one generated using CtoD from Raygui 4.0

What's strange is I tried using CtoD on both version 3.5 and version 4.0 of `raygui.h`, and the results are very different. On 3.5 it knew what if was doing. On 4.0 it's left way too many things unchanged from the C version.

Perhaps there was some newly-added symbol early in the file that made it trip.
